src/theme/publishTheme/hooks.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import { dark } from "./darkTheme";
+import { forest } from "./forestTheme";
+import { light } from "./lightTheme";
+import { PublishTheme } from "./publishTheme";
+
+export type ReTheme = {
+  themeType?: "light" | "dark" | "forest" | "custom";
+  themeTextColor?: string;
+  themeSelectColor?: string;
+  themeBackgroundColor?: string;
+};
+
+export default (seneThemeOptions?: ReTheme) => {
+  const [publishedTheme, setPublishedTheme] = useState<PublishTheme>(dark);
+  function addAlpha(color?: string, opacity?: number): string {
+    // coerce values so ti is between 0 and 1.
+    const _opacity = Math.round(Math.min(Math.max(opacity || 1, 0), 1) * 255);
+    return color + _opacity.toString(16).toUpperCase();
+  }
+
+  function isDark(color: string) {
```

Comment:
Check `atoms/Markdown/index.tsx` because it is using the tinycolor package already to check if a color is dark or not. 
Also check that package if it can be used to do any of the other things you are doing here manually!